pkg/rancher-desktop/main/deploymentProfiles.ts

```diff
@@ -52,38 +55,32 @@ export async function readDeploymentProfiles(): Promise<settings.DeploymentProfi
 
   switch (os.platform()) {
   case 'win32':
-    for (const key of [nativeReg.HKLM, nativeReg.HKCU]) {
-      const registryKey = nativeReg.openKey(key, REGISTRY_PATH_PROFILE.join('\\'), nativeReg.Access.READ);
+    // eslint-disable-next-line no-labels
+    findWin32Profile: {
+      for (const key of [nativeReg.HKLM, nativeReg.HKCU]) {
+        for (const registryPath of REGISTRY_PATH_PROFILE) {
+          const registryKey = nativeReg.openKey(key, registryPath.join('\\'), nativeReg.Access.READ);
+
+          if (!registryKey) {
+            continue;
+          }
 
-      if (!registryKey) {
-        continue;
-      }
-      const defaultsKey = nativeReg.openKey(registryKey, 'Defaults', nativeReg.Access.READ);
-      const lockedKey = nativeReg.openKey(registryKey, 'Locked', nativeReg.Access.READ);
+          if (!defaults || Object.keys(defaults).length === 0) {
+            defaults = readRegistryByType(registryKey, 'Defaults');
+          }
 
-      try {
-        if (defaultsKey) {
-          defaults = readRegistryUsingSchema(settings.defaultSettings, defaultsKey) ?? {};
-        }
-        if (lockedKey) {
-          locked = readRegistryUsingSchema(settings.defaultSettings, lockedKey) ?? {};
-        }
-      } catch (err) {
-        console.error( `Error reading deployment profile: ${ err }`);
-      } finally {
-        nativeReg.closeKey(registryKey);
-        if (defaultsKey) {
-          nativeReg.closeKey(defaultsKey);
-        }
-        if (lockedKey) {
-          nativeReg.closeKey(lockedKey);
+          if (!locked || Object.keys(locked).length === 0) {
+            locked = readRegistryByType(registryKey, 'Locked');
+          }
+
+          nativeReg.closeKey(registryKey);
+
+          if ((defaults && Object.keys(defaults).length) && (locked && Object.keys(locked).length)) {
```

Comment:
This should be `if ((defaults && Object.keys(defaults).length) || (locked && Object.keys(locked).length)) {`

Reply:
Same comment as above about issue #4518 saying to do it the way it's written.